extensions/observability-devservices/testcontainers/src/main/java/io/quarkus/observability/testcontainers/LgtmContainer.java

```diff
@@ -82,7 +119,7 @@ private String getPrometheusConfig() {
         Config runtimeConfig = ConfigProvider.getConfig();
         String rootPath = runtimeConfig.getOptionalValue("quarkus.management.root-path", String.class).orElse("/q");
         String metricsPath = runtimeConfig.getOptionalValue("quarkus.management.metrics.path", String.class).orElse("/metrics");
-        int httpPort = runtimeConfig.getOptionalValue("quarkus.http.port", Integer.class).orElse(0);
+        int httpPort = runtimeConfig.getOptionalValue("quarkus.http.port", Integer.class).orElse(8080); // when not set use default
```

Comment:
Doesn't the 0 do that -- helps set the default?

Reply:
The problem is that `0` is the default and should mean some random port, however, in devmode, the default port is `8080`.
Unless the property is set by the user, the port ended up being set to `0` in the scraper, leading to no data collection. An insidious bug.